### PR TITLE
feat(client): flink application support

### PR DESCRIPTION
**Pull request by Serpentiel**
adds support for Flink Application

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -14,6 +14,28 @@ type APIResponse struct {
...
+}
+
+// CreationSignatures represents a set of fields that are used to track the creation.
+type CreationSignatures struct {
+	CreatedAt string `json:"created_at,omitempty"`
+	CreatedBy string `json:"created_by.omitempty"`
+}
+
+// UpdationSignatures represents a set of fields that are used to track the updates.
+type UpdationSignatures struct {
```
Comment (mhoffm-aiven) [inline api.go:34]: Would go with `UpdateSignatures` maybe

**Comment by mhoffm-aiven** [comment]
Is there no schema currently for the `interface{}` values?

**Comment by Serpentiel** [comment]
> Is there no schema currently for the `interface{}` values?

there is, there are no plans to work with that schema in our IaaC tools, this could be added in future if there is need for it, but this is quite urgent at the moment

File: api.go [hunk 2]
```diff
@@ -14,6 +14,28 @@ type APIResponse struct {
 	Message string  `json:"message,omitempty"`
 }
 
+// Identifiable represents a resource with an ID.
+type Identifiable struct {
```
Comment (ivan-savciuc) [inline api.go:18]: Can we make a Flink similar to all the other API wrappers? 
Moreover something like this: 
```
CreateFlinkApplicationRequest{
		Nameable: Nameable{
			Name: name.
		},
	}
```
does not look good  at all.

File: flink_application.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+) (*DetailedFlinkApplicationResponse, error) {
+	path := buildPath("project", project, "service", service, "flink", "application", applicationID)
+
+	bts, err := h.client.doGetRequest(path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var r DetailedFlinkApplicationResponse
+
```
Comment (ivan-savciuc) [inline flink_application.go:68]: I this these new line break is no necessary here between two connected statements

**Review by ivan-savciuc — CHANGES_REQUESTED** [review]
Nitpicking: some style related comments otherwise LGTM

**Review by ivan-savciuc — APPROVED** [review]
LGTM

**Pull request merged**